### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by blase147**
In this project we;

- Added unit tests for all the methods in all the classes:
  - Book
  - Person
  - Student
  - Teacher
  - Classroom
  - Rental
  - all classes that we created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require 'spec_helper'
+
+describe Person do
```
Comment (rustam-tolipov) [inline spec/person_spec.rb:3]: - [ ] Your tests look great. But need to look deeper into this project. So, to test your cases I suggest you create a dynamic date like this: 

```
book = Book.new("Title", "Author")
person = Person.new(24, "Bob")
```

File: spec/student_spec.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require 'spec_helper'
+
+describe Student do
+  before :each do
+    @student = Student.new 'Title', 'Author'
```
Comment (rustam-tolipov) [inline spec/student_spec.rb]: - [ ] You also need to write tests for positive and negative scenarios. For example, you have parental permission. Or can use services. So you can decide which of them you choose. But you must have written tests for both scenarios. Kindly add them.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello Team,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — APPROVED** [review]
Hi @blase147, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
